### PR TITLE
GUM-27: Create onboarding screen

### PR DESCRIPTION
**Pull request by katyakostikova**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,11 +22,14 @@
     "react-native": "0.67.2",
     "react-native-config": "^1.4.5",
     "react-native-encrypted-storage": "^4.0.2",
-    "react-native-fingerprint-scanner": "^6.0.0",
+		"react-native-fingerprint-scanner": "^6.0.0",
```
Comment (R-Mielamud) [inline package.json]: Better remove this long tabulation. Maybe it's a TAB of size 2. Use 2 SPACES instead.

File: src/common/enums/navigation/app-route.enum.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ enum AppRoute {
   OKR = 'OKR',
   PROFILE = 'Profile',
   APP = 'App',
+  ONBOARDING_SETUP = 'Onboarding setup',
```
Comment (R-Mielamud) [inline src/common/enums/navigation/app-route.enum.ts]: Don't use spaces in these identifiers. Better write `OnboardingSetup`.

File: src/common/enums/navigation/onboarding-route.enum.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+enum OnboardingRoute {
+  ONBOARDING = 'Onboarding',
+  ADD_EXPERIENCE = 'Add experience',
+  ADD_EDUCATION = 'Add education',
+  ADD_LANGUAGE = 'Add language',
+  ADD_LOCATION = 'Add location',
```
Comment (R-Mielamud) [inline src/common/enums/navigation/onboarding-route.enum.ts]: Same here. PascalCase instead of     spaces

File: src/components/input/date-input.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import { AppColor } from 'src/common/enums';
+import styles from './styles';
+
+type InputProps = React.ComponentPropsWithoutRef<typeof DatePickerInput>;
+
+const DateInput: React.FC<InputProps> = ({
+  label,
+  style,
+
```
Comment (R-Mielamud) [inline src/components/input/date-input.tsx]: Remove new line.

File: package.json (hunk 1 above)
Comment (AleksandrTermenzhy) [inline package.json]: ?

File: src/screens/add-education/add-education.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import { useNavigation } from '@react-navigation/native';
+import React, { useCallback, useState } from 'react';
+import { ScrollView, View } from 'react-native';
+import { en, registerTranslation } from 'react-native-paper-dates';
+
+import { ButtonMode, HeadingLevel } from 'src/common/enums';
+import { DateInput, Heading, Input, MainButton } from 'src/components';
+import styles from './styles';
+
+type AddEducationScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/add-education/add-education.tsx]: Why `never`? This thing isn't an error. Better define an interface/object type:
```typescript
type AddEducationScreenProps = {...}
```

File: src/screens/add-education/add-education.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+type Period = {
+  startDate: Date | undefined;
+  endDate: Date | undefined;
+};
+
+registerTranslation('en', en);
```
Comment (R-Mielamud) [inline src/screens/add-education/add-education.tsx]: Is this line related to `AddEducationScreen`? Explain if it is or move it `src/config/i18n.ts`, `configureI18N` function, which is called in `src/config/index.ts` in function `configureApp` and `configureApp` is called in `index.js`.

Like this:
```typescript
# src/config/i18n.ts

const configureI18N = () => {
  registerTranslation('en', en);
};

# src/config/index.ts

const configureApp = () => {
  configureI18N();
};

# index.js

configureApp();
...register...
```

File: src/components/input/date-input.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { DatePickerInput } from 'react-native-paper-dates';
+
+import { AppColor } from 'src/common/enums';
+import styles from './styles';
+
+type InputProps = React.ComponentPropsWithoutRef<typeof DatePickerInput>;
```
Comment (AleksandrTermenzhy) [inline src/components/input/date-input.tsx]:
```suggestion
type DateInputProps = React.ComponentPropsWithoutRef<typeof DatePickerInput>;
```

File: src/screens/add-education/add-education.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+  return (
+    <View style={styles.screen}>
+      <ScrollView showsVerticalScrollIndicator={false}>
+        <View>
+          <View style={styles.inputContent}>
+            <Heading style={styles.heading} level={HeadingLevel.H5}>
+              University
+            </Heading>
+            <Input placeholder="University Name" />
```
Comment (R-Mielamud) [inline src/screens/add-education/add-education.tsx]: Maybe this View with Inputs should wrapped in Form

Comment (R-Mielamud) [reply src/screens/add-education/add-education.tsx]: It it's too early to do this, add a TODO comment

File: src/screens/onboarding/components/step-content/education-content.tsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { useNavigation } from '@react-navigation/native';
+import React, { useCallback } from 'react';
+import { ScrollView, View } from 'react-native';
+
+import { HeadingLevel, OnboardingRoute } from 'src/common/enums';
+import { Heading } from 'src/components';
+import { AddButton } from '..';
+import styles from './styles';
+
+type EducationContentScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/onboarding/components/step-content/education-content.tsx]: Again, why never? We use TypeScript to avoid such things :)
If it means 'no props', just remove `React.FC`'s type parameter

File: src/components/input/date-input.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+const DateInput: React.FC<InputProps> = ({
+  label,
+  style,
+
+  ...textInputProps
```
Comment (AleksandrTermenzhy) [inline src/components/input/date-input.tsx]:
```suggestion
  style,
  ...textInputProps
```

File: src/screens/add-experience/add-experience.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+import React, { useCallback, useState } from 'react';
+import { View } from 'react-native';
+import { en, registerTranslation } from 'react-native-paper-dates';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import { ButtonMode, HeadingLevel } from 'src/common/enums';
+import { DateInput, Heading, Input, MainButton } from 'src/components';
+import styles from './styles';
+
+type AddExperienceScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/add-experience/add-experience.tsx]: Same here about never.

File: src/components/input/date-input.tsx [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+}) => {
+  return (
+    <DatePickerInput
+      style={[styles.inputContainer, style]}
+      label={label}
```
Comment (AleksandrTermenzhy) [inline src/components/input/date-input.tsx]:
```suggestion
```

File: src/screens/add-experience/add-experience.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+type Period = {
+  startDate: Date | undefined;
+  endDate: Date | undefined;
+};
+
+registerTranslation('en', en);
```
Comment (R-Mielamud) [inline src/screens/add-experience/add-experience.tsx]: Move registerTranslation to `src/config` instead of calling in every file

File: src/screens/add-language/add-language.tsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import { useNavigation } from '@react-navigation/native';
+import React, { useCallback } from 'react';
+import { View } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import { ButtonMode, HeadingLevel } from 'src/common/enums';
+import { Heading, Input, MainButton } from 'src/components';
+import styles from './styles';
+
+type AddLanguageScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/add-language/add-language.tsx]: Again never

File: src/screens/onboarding/components/step-content/experience-content.tsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
+import { useNavigation } from '@react-navigation/native';
+import React, { useCallback } from 'react';
+import { ScrollView, View } from 'react-native';
+
+import { HeadingLevel, OnboardingRoute } from 'src/common/enums';
+import { Heading } from 'src/components';
+import { AddButton } from '../';
+import styles from './styles';
+
+type ExperienceContentScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/onboarding/components/step-content/experience-content.tsx]: Again never.

File: src/screens/onboarding/components/step-content/interesting-content.tsx [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+import { en, registerTranslation } from 'react-native-paper-dates';
+
+import { HeadingLevel, IconName, OnboardingRoute } from 'src/common/enums';
+import { Heading } from 'src/components';
+import { AddButton } from '..';
+import styles from './styles';
+
+type InterestingContentScreenProps = Record<string, never>;
+
+registerTranslation('en', en);
```
Comment (R-Mielamud) [inline src/screens/onboarding/components/step-content/interesting-content.tsx]: Never in props + move translation to config file

File: src/screens/onboarding/components/step-content/user-content.tsx [hunk 17]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { View } from 'react-native';
+import { Avatar } from 'react-native-paper';
+
+import { HeadingLevel } from 'src/common/enums';
+import { Heading, Input } from 'src/components';
+import styles from './styles';
+
+type UserContentScreenProps = Record<string, never>;
```
Comment (R-Mielamud) [inline src/screens/onboarding/components/step-content/user-content.tsx]: Again never

File: src/components/main-button/main-button.tsx [hunk 18]
```diff
@@ -8,13 +8,19 @@ type MainButtonProps = React.ComponentPropsWithoutRef<typeof Button>;
...
 }) => {
   return (
     <Button
       uppercase={false}
-      style={mode === ButtonMode.OUTLINED && styles[ButtonMode.OUTLINED]}
+      labelStyle={[styles.text, labelStyle]}
+      style={[
+        mode === ButtonMode.OUTLINED && styles[ButtonMode.OUTLINED],
+        style,
+      ]}
```
Comment (AleksandrTermenzhy) [inline src/components/main-button/main-button.tsx]: Let's take this to separate variable for the sake of readability and wrap into `useMemo()`

File: src/screens/onboarding/components/step-dots/step-dots.tsx [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+  activeDot: number;
+};
+
+const StepDots: React.FC<StepDotsProps> = ({ activeDot }) => {
+  return (
+    <View style={styles.dotsContainer}>
+      <View style={[styles.dot, activeDot === 0 && styles.activeDot]} />
+      <View style={[styles.dot, activeDot === 1 && styles.activeDot]} />
+      <View style={[styles.dot, activeDot === 2 && styles.activeDot]} />
+      <View style={[styles.dot, activeDot === 3 && styles.activeDot]} />
```
Comment (R-Mielamud) [inline src/screens/onboarding/components/step-dots/step-dots.tsx]: Move this thing (`activeDot === 3 && styles.activeDot`) to separate `getActiveStyle` function, which can be implemented like this:
```typescript
const getActiveStyle = (dotIndex: number) => {
  if (activeDot === dotIndex) {
    return styles.activeDot;
  }

  return {};
};
```

File: src/screens/onboarding/onboarding.tsx [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+          <View key="2">
+            <ExperienceContent />
+          </View>
+          <View key="3">
+            <EducationContent />
+          </View>
+          <View key="4">
+            <InterestingContent />
+          </View>
+        </PagerView>
```
Comment (R-Mielamud) [inline src/screens/onboarding/onboarding.tsx]: Keys are necessary only inside `Array.map` or something like that.
But if there's a warning during build, leave it like this.

File: src/screens/onboarding/onboarding.tsx [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+        <View style={styles.buttonContainer}>
+          <StepDots activeDot={currentStep} />
+          {currentStep === 3 ? (
+            <MainButton
+              mode={ButtonMode.CONTAINED}
+              onPress={() => pagerRef.current?.setPage(currentStep + 1)}
```
Comment (R-Mielamud) [inline src/screens/onboarding/onboarding.tsx]: Extract a separate method

File: src/screens/onboarding/onboarding.tsx [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+            >
+              Complete
+            </MainButton>
+          ) : (
+            <MainButton
+              mode={ButtonMode.OUTLINED}
+              onPress={() => pagerRef.current?.setPage(currentStep + 1)}
```
Comment (R-Mielamud) [inline src/screens/onboarding/onboarding.tsx]: Same here

**Review by R-Mielamud — CHANGES_REQUESTED** [review]

File: src/navigation/onboarding/onboarding-navigation.tsx [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+        component={AddEducationScreen}
+      />
+      <Stack.Screen
+        name={OnboardingRoute.ADD_LANGUAGE}
+        component={AddLanguageScreen}
+      />
+      <Stack.Screen
+        name={OnboardingRoute.ADD_LOCATION}
+        component={AddLocationScreen}
+      />
```
Comment (AleksandrTermenzhy) [inline src/navigation/onboarding/onboarding-navigation.tsx]: Let's wrap this `add`-screens into `Stack.Group` with `presentation: 'modal'` option. This should improve UX for such secondary screens.

File: src/screens/add-education/add-education.tsx [hunk 24]
```diff
@@ -0,0 +1,89 @@
...
+    startDate: undefined,
+    endDate: undefined,
+  });
+
+  const handleCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  return (
+    <View style={styles.screen}>
```
Comment (AleksandrTermenzhy) [inline src/screens/add-education/add-education.tsx]:
```suggestion
    <SafeAreaView style={styles.screen}>
```

File: src/screens/add-education/add-education.tsx [hunk 25]
```diff
@@ -0,0 +1,89 @@
...
+            </Heading>
+            <Input placeholder="Degree" />
+          </View>
+          <View style={styles.inputContent}>
+            <Heading style={styles.heading} level={HeadingLevel.H5}>
+              Period
+            </Heading>
+            <DateInput
+              label="Start Date"
+              onChange={date => setPeriod(curr => ({ ...curr, endDate: date }))}
```
Comment (AleksandrTermenzhy) [inline src/screens/add-education/add-education.tsx]: Please take to the `handleStartDateChange()`

File: src/screens/add-education/add-education.tsx [hunk 26]
```diff
@@ -0,0 +1,89 @@
...
+              value={period.startDate}
+              locale="en"
+              inputMode="start"
+            />
+            <DateInput
+              onChange={date =>
+                setPeriod(curr => ({ ...curr, startDate: date }))
```
Comment (AleksandrTermenzhy) [inline src/screens/add-education/add-education.tsx]: same here

File: src/screens/add-education/add-education.tsx (hunk 6 above)
Comment (AleksandrTermenzhy) [inline src/screens/add-education/add-education.tsx]: We should move this to `App.tsx` to be called only once for entire project.

File: src/screens/add-education/styles.ts [hunk 27]
```diff
@@ -0,0 +1,27 @@
+import { StyleSheet } from 'react-native';
+
+import { AppColor } from 'src/common/enums';
+
+const styles = StyleSheet.create({
+  screen: {
+    padding: 20,
+    backgroundColor: AppColor.WHITE,
+    justifyContent: 'space-between',
+    height: '100%',
```
Comment (AleksandrTermenzhy) [inline src/screens/add-education/styles.ts]: Generally `flex: 1` should be enough. If there is a need to add `height: 100% ` explicitly, maybe something's wrong with layout

File: src/screens/add-language/add-language.tsx [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+const AddLanguageScreen: React.FC<AddLanguageScreenProps> = () => {
+  const navigation = useNavigation();
+
+  const handleCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  return (
+    <SafeAreaView>
+      <View style={styles.screen}>
```
Comment (AleksandrTermenzhy) [inline src/screens/add-language/add-language.tsx]:
```suggestion
    <SafeAreaView style={styles.screen}>
```

File: src/screens/add-experience/add-experience.tsx (hunk 13 above)
Comment (AleksandrTermenzhy) [inline src/screens/add-experience/add-experience.tsx]: Everything for education screen applies here as well.

File: src/screens/add-location/add-location.tsx [hunk 29]
```diff
@@ -0,0 +1,41 @@
...
+const AddLocationScreen: React.FC<AddLocationScreenProps> = () => {
+  const navigation = useNavigation();
+
+  const handleCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  return (
+    <SafeAreaView>
+      <View style={styles.screen}>
```
Comment (AleksandrTermenzhy) [inline src/screens/add-location/add-location.tsx]: We don't really need two nested views here.

File: src/screens/onboarding/components/step-content/styles.ts [hunk 30]
```diff
@@ -0,0 +1,40 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  input: {
+    marginBottom: 20,
+    width: '90%',
+    alignSelf: 'center',
+  },
+  content: {
+    width: '90%',
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/components/step-content/styles.ts]: Let's introduce some shared constant for default screen padding instead.

File: src/screens/onboarding/components/step-content/user-content.tsx [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+const UserContent: React.FC<UserContentScreenProps> = () => {
+  return (
+    <View style={styles.content}>
+      <Heading style={styles.header} level={HeadingLevel.H4}>
+        Tell Us About Yourself
+      </Heading>
+      <View style={styles.imageContainer}>
+        <Avatar.Image
+          size={150}
+          source={require('src/assets/images/user-icon.jpeg')}
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/components/step-content/user-content.tsx]: Can we use icon instead of file?

File: src/screens/onboarding/components/step-dots/step-dots.tsx [hunk 32]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { View } from 'react-native';
+
+import styles from './styles';
+
+type StepDotsProps = {
+  activeDot: number;
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/components/step-dots/step-dots.tsx]:
```suggestion
  activeIndex: number;
```

File: src/screens/onboarding/components/step-dots/step-dots.tsx (hunk 19 above)
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/components/step-dots/step-dots.tsx]: I'd suggest passing dots count as a prop and use `map` here.

File: src/screens/onboarding/onboarding.tsx [hunk 33]
```diff
@@ -0,0 +1,68 @@
...
+
+type IOnboardingScreenProps = Record<string, never>;
+
+const OnboardingScreen: React.FC<IOnboardingScreenProps> = () => {
+  const [currentStep, setCurrentStep] = useState(0);
+  const pagerRef = useRef<PagerView | null>(null);
+
+  return (
+    <SafeAreaView>
+      <View style={styles.screen}>
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]: You can skip `View`

File: src/screens/onboarding/onboarding.tsx [hunk 34]
```diff
@@ -0,0 +1,68 @@
...
+        <PagerView
+          style={styles.content}
+          initialPage={0}
+          ref={pagerView => (pagerRef.current = pagerView)}
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]:
```suggestion
          ref={pagerRef}
```

File: src/screens/onboarding/onboarding.tsx [hunk 35]
```diff
@@ -0,0 +1,68 @@
...
+          onPageScroll={e => setCurrentStep(e.nativeEvent.position)}
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]: Please take this to a separate `handlePageScroll` function

File: src/screens/add-education/add-education.tsx (hunk 5 above)
Comment (katyakostikova) [reply src/screens/add-education/add-education.tsx]: image

File: src/screens/onboarding/onboarding.tsx [hunk 36]
```diff
@@ -0,0 +1,68 @@
...
+        >
+          <View key="1">
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]: We might need to add `collapsible={false}`

File: src/screens/onboarding/onboarding.tsx [hunk 37]
```diff
@@ -0,0 +1,68 @@
...
+          <View key="3">
+            <EducationContent />
+          </View>
+          <View key="4">
+            <InterestingContent />
+          </View>
+        </PagerView>
+        <View style={styles.buttonContainer}>
+          <StepDots activeDot={currentStep} />
+          {currentStep === 3 ? (
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]: I'd suggest introducing `isLastStep` variable.

File: src/screens/onboarding/styles.ts [hunk 38]
```diff
@@ -0,0 +1,20 @@
+import { StyleSheet } from 'react-native';
+
+import { AppColor } from 'src/common/enums';
+
+const styles = StyleSheet.create({
+  screen: {
+    backgroundColor: AppColor.WHITE,
+  },
+  content: {
+    height: '83%',
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/styles.ts]: What are these magic-numbers?

File: src/screens/add-education/add-education.tsx (hunk 5 above)
Comment (R-Mielamud) [reply src/screens/add-education/add-education.tsx]: Then just write `React.FC` instead of `React.FC<AddEducationProps>`

File: src/navigation/index.tsx [hunk 39]
```diff
@@ -23,7 +23,10 @@ const RootNavigation = () => {
         <RootStack.Screen name={AppRoute.APP} component={AppNavigation} />
       ) : (
         <>
-          <RootStack.Screen name={AuthRoute.SIGN_IN} component={HomeScreen} />
+          <RootStack.Screen
+            name={AuthRoute.SIGN_IN}
+            component={OnboardingNavigation}
```
Comment (AleksandrTermenzhy) [inline src/navigation/index.tsx]: Onboarding should be within `app`-routes, not `auth`

File: src/screens/onboarding/components/step-dots/step-dots.tsx [hunk 40]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { View } from 'react-native';
+
+import styles from './styles';
+
+type StepDotsProps = {
+  activeIndex: number;
+  count: number;
+};
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/components/step-dots/step-dots.tsx]:
```suggestion
interface IStepDotsProps {
  activeIndex: number;
  count: number;
};
```

File: src/screens/onboarding/onboarding.tsx [hunk 41]
```diff
@@ -0,0 +1,80 @@
...
+  InterestingContent,
+  StepDots,
+  UserContent,
+} from './components';
+import styles from './styles';
+
+type IOnboardingScreenProps = Record<string, never>;
+
+const numberOfDots = 4;
+const lastStepIndex = 3;
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]:
```suggestion
const ONBOARDING_DOTS_COUNT = 4;
const LAST_STEP_INDEX = ONBOARDING_DOTS_COUNT - 1;
```

File: src/screens/onboarding/onboarding.tsx [hunk 42]
```diff
@@ -0,0 +1,80 @@
...
+
+  const changeCurrentPage = () => {
+    pagerRef.current?.setPage(currentStep + 1);
+  };
+
+  const handlePageScroll = useCallback(
+    (e: PagerViewOnPageScrollEvent) => {
+      setCurrentStep(e.nativeEvent.position);
+    },
+    [setCurrentStep]
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/onboarding.tsx]:
```suggestion
    []
```

It's safe to omit `useState` setters.

File: src/screens/onboarding/styles.ts [hunk 43]
```diff
@@ -0,0 +1,26 @@
...
+
+const styles = StyleSheet.create({
+  screen: {
+    backgroundColor: AppColor.WHITE,
+    flex: 1,
+    justifyContent: 'space-between',
+  },
+  content: {
+    width: '100%',
+    height: '80%',
```
Comment (AleksandrTermenzhy) [inline src/screens/onboarding/styles.ts]: Why do we need this?

File: src/navigation/index.tsx (hunk 39 above)
Comment (katyakostikova) [reply src/navigation/index.tsx]: I just forgot to remove onborading from navigation, as we need LoginScreen here, I added it here just for testing

File: src/screens/add-education/add-education.tsx [hunk 44]
```diff
@@ -0,0 +1,97 @@
...
+    (date: Date | undefined) => {
+      setPeriod(curr => ({ ...curr, endDate: date }));
+    },
+    [setPeriod]
+  );
+
+  return (
+    <SafeAreaView style={styles.screen}>
+      <ScrollView showsVerticalScrollIndicator={false}>
+        <View>
```
Comment (R-Mielamud) [inline src/screens/add-education/add-education.tsx]: Wrap with Formik (use blank initial values, etc., but Formik must be here) and use FormInput, FormSelect, etc.
\+ Create FormDateInput

File: src/screens/add-experience/add-experience.tsx [hunk 45]
```diff
@@ -0,0 +1,94 @@
...
+    [setPeriod]
+  );
+
+  return (
+    <SafeAreaView style={styles.screen}>
+      <ScrollView
+        contentContainerStyle={styles.content}
+        showsVerticalScrollIndicator={false}
+      >
+        <View>
```
Comment (R-Mielamud) [inline src/screens/add-experience/add-experience.tsx]: Same here about form

File: src/screens/add-language/add-language.tsx [hunk 46]
```diff
@@ -0,0 +1,39 @@
...
+const AddLanguageScreen: React.FC = () => {
+  const navigation = useNavigation();
+
+  const handleCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  return (
+    <SafeAreaView style={styles.screen}>
+      <View style={styles.inputContent}>
```
Comment (R-Mielamud) [inline src/screens/add-language/add-language.tsx]: Also form

File: src/screens/add-location/add-location.tsx [hunk 47]
```diff
@@ -0,0 +1,39 @@
...
+const AddLocationScreen: React.FC = () => {
+  const navigation = useNavigation();
+
+  const handleCancel = useCallback(() => {
+    navigation.goBack();
+  }, [navigation]);
+
+  return (
+    <SafeAreaView style={styles.screen}>
+      <View style={styles.inputContent}>
```
Comment (R-Mielamud) [inline src/screens/add-location/add-location.tsx]: Same here

File: src/screens/onboarding/components/step-content/user-content.tsx [hunk 48]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <View style={styles.content}>
+      <ScrollView showsVerticalScrollIndicator={false}>
+        <Heading style={styles.header} level={HeadingLevel.H4}>
+          Tell Us About Yourself
+        </Heading>
+        <View style={styles.imageContainer}>
+          <Avatar.Icon
+            style={styles.userAvatar}
+            size={150}
```
Comment (R-Mielamud) [inline src/screens/onboarding/components/step-content/user-content.tsx]: This is like `fontSize`. For such big values use multiples of 8. For example `152`

**Review by R-Mielamud — COMMENTED** [review]
Some minor fixes

File: src/screens/onboarding/components/step-content/user-content.tsx (hunk 48 above)
Comment (katyakostikova) [reply src/screens/onboarding/components/step-content/user-content.tsx]: Could you please explain why?

Comment (R-Mielamud) [reply src/screens/onboarding/components/step-content/user-content.tsx]: From Figma docs:
> In the 8-point grid system, everything is spaced and sized using numbers that are multiples of 8. Designers will often use an 8-point grid for spacing and sizing with a [4-point baseline grid](https://material.io/design/layout/spacing-methods.html#baseline) for typography. The type baselines will sit on on the 4-point grid. In order to achieve this, the focus here is really more on the line-heights than the exact font sizes.

Comment (R-Mielamud) [reply src/screens/onboarding/components/step-content/user-content.tsx]: And you have an icon there

File: src/screens/add-education/add-education.tsx (hunk 44 above)
Comment (katyakostikova) [reply src/screens/add-education/add-education.tsx]: I've also added validation schemas

**Comment by katyakostikova** [comment]
I've added validation schemas and formik

File: src/components/main-button/main-button.tsx [hunk 49]
```diff
@@ -8,13 +8,18 @@ type MainButtonProps = React.ComponentPropsWithoutRef<typeof Button>;
 
 const MainButton: React.FC<MainButtonProps> = ({
   children,
+  style,
+  labelStyle,
   mode = ButtonMode.TEXT,
   ...buttonProps
 }) => {
+  const isOutlined = useMemo(() => mode === ButtonMode.OUTLINED, [mode]);
```
Comment (liza-veis) [inline src/components/main-button/main-button.tsx]: We don't need useMemo here. We should use useMemo either for complex calculations or if you don't want to re-create object on every rerender

File: src/screens/add-education/add-education.tsx [hunk 50]
```diff
@@ -0,0 +1,96 @@
...
+  courseName: '',
+  degree: '',
+  startDate: undefined,
+  endDate: undefined,
+};
+
+const AddEducationScreen: React.FC = () => {
+  const navigation = useNavigation();
+
+  const handleCancel = useCallback(() => {
```
Comment (liza-veis) [inline src/screens/add-education/add-education.tsx]: In such cases it's better not to use useCallback, remove it from other places too. Please take a look at this [article](https://kentcdodds.com/blog/usememo-and-usecallback) when you have time

File: src/screens/add-education/add-education.tsx [hunk 51]
```diff
@@ -0,0 +1,104 @@
...
+import { ScrollView, View } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { Formik } from 'formik';
+
+import { ButtonMode, HeadingLevel } from 'src/common/enums';
+import { FormDate, FormInput, Heading, MainButton } from 'src/components';
+import { educationValidationSchema } from 'src/validation-schemas';
+import styles from './styles';
+
+const initialValues = {
```
Comment (liza-veis) [inline src/screens/add-education/add-education.tsx]: Let's rename all initialValues to something like defaultAddEducationPayload and move to common.ts file as it's done for CreateSkillForm

File: src/screens/add-education/add-education.tsx [hunk 52]
```diff
@@ -0,0 +1,104 @@
...
+                  </Heading>
+                  <FormDate
+                    label="Start Date"
+                    locale="en"
+                    inputMode="start"
+                    name="startDate"
+                  />
+                  <FormDate
+                    style={styles.dateInput}
+                    locale="en"
```
Comment (liza-veis) [inline src/screens/add-education/add-education.tsx]: Let's set locale for DateInput by default

File: src/screens/add-education/add-education.tsx [hunk 53]
```diff
@@ -0,0 +1,104 @@
...
+        <Formik
+          validationSchema={educationValidationSchema}
+          initialValues={initialValues}
+          onSubmit={() => {
+            // TODO
+          }}
+        >
+          {({ isValid, handleSubmit }) => (
+            <>
+              <View>
```
Comment (liza-veis) [inline src/screens/add-education/add-education.tsx]:
```suggestion
```

File: src/screens/add-experience/add-experience.tsx [hunk 54]
```diff
@@ -0,0 +1,97 @@
...
+        <Formik
+          validationSchema={experienceValidationSchema}
+          initialValues={initialValues}
+          onSubmit={() => {
+            // TODO
+          }}
+        >
+          {({ isValid, handleSubmit }) => (
+            <>
+              <View>
```
Comment (liza-veis) [inline src/screens/add-experience/add-experience.tsx]:
```suggestion
```

File: src/screens/onboarding/components/step-dots/step-dots.tsx [hunk 55]
```diff
@@ -0,0 +1,29 @@
...
+
+    return {};
+  };
+
+  return (
+    <View style={styles.dotsContainer}>
+      {[...Array(count)].map((_, index) => (
+        <View key={index} style={[styles.dot, getActiveStyle(index)]} />
+      ))}
+    </View>
```
Comment (liza-veis) [inline src/screens/onboarding/components/step-dots/step-dots.tsx]: I think it's better to write it like this
```suggestion
    <View style={styles.dotsContainer}>
      {[...Array(count)].map((_, index) => {
        const isActive = activeIndex === index;

        return (
          <View key={index} style={[styles.dot, isActive && styles.activeDot]} />
        );
      })}
    </View>
```

File: src/components/input/date-input.tsx [hunk 56]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (liza-veis) [inline src/components/input/date-input.tsx]: Move DateInput to it's own component

File: src/screens/onboarding/components/step-content/user-content.tsx [hunk 57]
```diff
@@ -0,0 +1,65 @@
...
+        <Formik
+          initialValues={initialValues}
+          validationSchema={userInfoValidationSchema}
+          onSubmit={() => {
+            // TODO
+          }}
+        >
+          <>
+            <FormInput
+              style={styles.input}
```
Comment (liza-veis) [inline src/screens/onboarding/components/step-content/user-content.tsx]: Use this style for containerStyle to avoid this 
Screenshot 2022-02-17 at 11 44 48

```suggestion
              containerStyle={styles.input}
```

File: src/styles/styles/global-styles.ts [hunk 58]
```diff
@@ -0,0 +1,3 @@
+const DEFAULT_SCREEN_PADDING = 20;
```
Comment (liza-veis) [inline src/styles/styles/global-styles.ts]: Let's rename styles folder inside styles to common

File: src/validation-schemas/education/education.ts [hunk 59]
```diff
@@ -0,0 +1,27 @@
+import * as yup from 'yup';
```
Comment (liza-veis) [inline src/validation-schemas/education/education.ts]: Rename this file to something like add-education.ts because we can have other validation schemas related to education

File: src/screens/onboarding/components/step-content/experience-content.tsx [hunk 60]
```diff
@@ -0,0 +1,39 @@
...
+import { useNavigation } from '@react-navigation/native';
+import { ScrollView, View } from 'react-native';
+
+import { HeadingLevel, OnboardingRoute } from 'src/common/enums';
+import { OnboardingStackParamList } from 'src/common/types';
+import { Heading } from 'src/components';
+import { AddButton } from '../';
+import styles from './styles';
+
+type ExperienceContentScreenProps = NativeStackNavigationProp<
```
Comment (liza-veis) [inline src/screens/onboarding/components/step-content/experience-content.tsx]: Do this for other components too
```suggestion
type ExperienceContentProps = NativeStackNavigationProp<
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by AleksandrTermenzhy — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Review by R-Mielamud — APPROVED** [review]
